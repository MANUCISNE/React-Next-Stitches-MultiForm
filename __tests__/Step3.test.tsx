import React from "react";
import Step3 from "../src/pages/step3";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

jest.mock("../src/contexts/FormStepContext", () => ({
  useGlobalContext: jest.fn(() => ({
    setAdditional: jest.fn(),
    setPlansMonth: jest.fn(),
    setPlansYearly: jest.fn(),
  })),
}));

describe('Step3 component', () => {
  test('renders yearly add-ons if package is true', () => {
    const { container } = render(<Step3 package={true} />);

    const yearlyAddons = screen.getByRole('listadd');

    expect(yearlyAddons).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders monthly add-ons if package is false', () => {
    const { container } = render(<Step3 package={false} />);

    const monthlyAddons = screen.getByRole('listitemcheckboxMO');

    expect(monthlyAddons).toBeInTheDocument()
    expect(container).toMatchSnapshot();
  });
});
