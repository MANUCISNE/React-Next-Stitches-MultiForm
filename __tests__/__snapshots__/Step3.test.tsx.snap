// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step3 component renders monthly add-ons if package is false 1`] = `
<div>
  <div
    class="c-jKOUQW"
  >
    <div
      class="c-clUunX"
      role="listadd"
    >
      <h2>
        Pick add-ons
      </h2>
      <p>
        Add-ons help enhance your gaming experience.
      </p>
      <div
        class="c-gCPNIP"
        role="listitemcheckboxMO"
      >
        <div
          class="c-eprAyq c-eprAyq-ijroWjL-css"
        >
          <button
            aria-checked="false"
            data-state="unchecked"
            id="1"
            role="checkbox"
            type="button"
            value="on"
          >
            <div
              style="display: flex; align-items: center; width: 1rem; height: 1rem; background: white; border-radius: 4px; border: 1px solid #c4c4cc;"
            />
          </button>
          <label
            class="c-bwiYgr c-bwiYgr-iBJYEN-css"
            for="c1"
          >
            <h4>
              Online service
            </h4>
            <p>
              Acess to multiplayer games
            </p>
          </label>
          <p
            style="color: rgb(0, 0, 255);"
          >
            $1/mo
          </p>
        </div>
        <div
          class="c-eprAyq c-eprAyq-ijroWjL-css"
        >
          <button
            aria-checked="false"
            data-state="unchecked"
            id="2"
            role="checkbox"
            type="button"
            value="on"
          >
            <div
              style="display: flex; align-items: center; width: 1rem; height: 1rem; background: white; border-radius: 4px; border: 1px solid #c4c4cc;"
            />
          </button>
          <label
            class="c-bwiYgr c-bwiYgr-iBJYEN-css"
            for="c1"
          >
            <h4>
              Advanced
            </h4>
            <p>
              Extra 1TB of could save your game
            </p>
          </label>
          <p
            style="color: rgb(0, 0, 255);"
          >
            $2/mo
          </p>
        </div>
        <div
          class="c-eprAyq c-eprAyq-ijroWjL-css"
        >
          <button
            aria-checked="false"
            data-state="unchecked"
            id="3"
            role="checkbox"
            type="button"
            value="on"
          >
            <div
              style="display: flex; align-items: center; width: 1rem; height: 1rem; background: white; border-radius: 4px; border: 1px solid #c4c4cc;"
            />
          </button>
          <label
            class="c-bwiYgr c-bwiYgr-iBJYEN-css"
            for="c1"
          >
            <h4>
              Pro
            </h4>
            <p>
              Custom theme on your profile
            </p>
          </label>
          <p
            style="color: rgb(0, 0, 255);"
          >
            $2/mo
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
